htdocs/bom/class/bom.class.php

```diff
@@ -1147,6 +1147,30 @@ public function getNetNeedsTree(&$TNetNeeds = array(), $qty = 0, $level = 0)
 			}
 		}
 	}
+
+	/**
+	 * Recursively retrieves all parent bom in the tree that leads to the $bom_id bom
+	 *
+	 * @param 	array	$TParentBom		We put all found parent bom in $TParentBom
+	 * @param 	int		$bom_id			ID of bom from which we want to get parent bom ids
+	 * @return 	void
+	 */
+	public function getParentBomTreeRecursive(&$TParentBom, $bom_id)
+	{
+
+		$sql = 'SELECT l.fk_bom, b.label
+				FROM '.MAIN_DB_PREFIX.'bom_bomline l
+				INNER JOIN '.MAIN_DB_PREFIX.$this->table_element.' b ON b.rowid = l.fk_bom
+				WHERE fk_bom_child = '.((int) $bom_id);
+
+		$resql = $this->db->query($sql);
+		if (!empty($resql)) {
+			while ($res = $this->db->fetch_object($resql)) {
```

Comment:
Recursive methods always need a protection against infinite loop.
Imagine data was loaded with a child thas has a child that is its parent, you fall into an infinite loop.
You must add a counter that is increased +1 at each call and we stop the recursion when counter > 1000 (we know the depth of BOM will never be 1000).

Reply:
Hi @eldy, i just did it